src/main/java/com/salesforce/dataloader/dyna/BooleanConverter.java

```diff
@@ -89,16 +89,16 @@ public BooleanConverter(Object defaultValue) {
     @Override
     public Object convert(Class type, Object value) {
 
-        if (value == null || String.valueOf(value).length()==0) {
-            return null;
+        if (value == null || value instanceof Boolean) {
+            return value;
         }
-
-        if (value instanceof Boolean) {
-            return (value);
+      
+        String stringValue = value.toString().trim();
+        if (stringValue.length()==0) {
```

Comment:
you can just do `value.toString().isEmpty()`

Reply:
Good point. However, given that we need to trim the string and use the trimmed substring to convert it to a Boolean later, I will keep it as-is.